src/Sylius/Bundle/CoreBundle/Applicator/CatalogPromotionApplicator.php

```diff
@@ -13,19 +13,39 @@
 
 namespace Sylius\Bundle\CoreBundle\Applicator;
 
+use Sylius\Component\Core\Model\CatalogPromotionInterface;
 use Sylius\Component\Core\Model\ChannelPricingInterface;
 use Sylius\Component\Core\Model\ProductVariantInterface;
+use Sylius\Component\Promotion\Model\CatalogPromotionTranslationInterface;
 
 final class CatalogPromotionApplicator implements CatalogPromotionApplicatorInterface
 {
-    public function applyPercentageDiscount(ProductVariantInterface $variant, float $discount): void
-    {
+    public function applyCatalogPromotion(
+        ProductVariantInterface $variant,
+        CatalogPromotionInterface $catalogPromotion
+    ): void {
         /** @var ChannelPricingInterface $channelPricing */
         foreach ($variant->getChannelPricings() as $channelPricing) {
             if ($channelPricing->getOriginalPrice() === null) {
                 $channelPricing->setOriginalPrice($channelPricing->getPrice());
             }
-            $channelPricing->setPrice((int) ($channelPricing->getPrice() - ($channelPricing->getPrice() * $discount)));
+
+            $channelPricing->setPrice((int) ($channelPricing->getPrice() - ($channelPricing->getPrice() * 0.5)));
+
+            $channelPricing->addAppliedPromotion($this->formatAppliedPromotion($catalogPromotion));
         }
     }
+
+    private function formatAppliedPromotion(CatalogPromotionInterface $catalogPromotion): array
```

Comment:
WDYT about moving this function to a separate service? if someone will want to add more information to channel pricing about catalog promotion, in this implementation their must overwrite the entire `CatalofPromotionApplicator` instead only `AppliedPromotionFormatter`
or maybe I missing something :D?